Sources/_CryptoExtras/RSA/RSA_boring.swift

```diff
@@ -233,6 +241,33 @@ extension BoringSSLRSAPublicKey {
                 return rc == 1
             }
         }
+        
+        fileprivate func encrypt<D: DataProtocol>(_ data: D, padding: _RSA.Encryption.Padding) throws -> _RSA.Encryption.RSAEncryptedData {
+            let outputSize = Int(CCryptoBoringSSL_RSA_size(self.pointer))
+
+            let output = try Array<UInt8>(unsafeUninitializedCapacity: outputSize) { bufferPtr, length in
```

Comment:
Pivoting through `Array` doesn't have the impact we want here, as we have to immediately allocate a `Data`. It'd be better to zero-initialize the `Data` and then use `withUnsafeMutableBytes`.

Reply:
This is another case where I was copy-pasting (from `signature()`, see below) and adapting to the purpose; doesn't it end up having the same problem, just slightly less explicitly?

Reply:
The existence of the RSASignature type is why the other side does this differently: given that we should just return a `Data` here, we can safely avoid the trouble.

Reply:
My kingdom for `Data.init(unsafeUninitializedCapacity:initializingWith:)` 😂